### ISSUE TITLE
suspect file-based is broken

### ISSUE THREAD

**Issue by BenjaminHae** [diafygi/letsencrypt-nosudo#31]
I'm not completely sure as I'm not whitelisted I couldn't test it. I believe when using the file-based challenge line 308

``` python
.format(n + 4, i['domain'], challenge['token'], responses[n]['data']))
```

will fail because `challenge` is not available in that context.

**Comment by sindarina** [comment]
It looks like file-based is indeed broken, probably because something changed in the way the challenge operates right now.

For example, this is what the script requests;

```
URL: http://www.example.com/.well-known/acme-challenge/5s69AbGkkhiWbXysZ42N_8CvVL1wUXkseB27JviXQDo
File contents: "Srg8yywjFeJs-MeHvRLjQ_Z4ZQMrqWsgAmPQIG3VLwQ.ClOPCDaVwJauu0ysYcc0pvxGyHfwI1LWg4Fq1SuYTJE"
```

And this is what the ACME validation server actually sends as a GET request;

```
"GET /.well-known/acme-challenge/Srg8yywjFeJs-MeHvRLjQ_Z4ZQMrqWsgAmPQIG3VLwQ HTTP/1.1"
```

Which is actually the first half (split on the dot) of what should be the file contents, and not the value that the script suggests. The URL that the script generates should be different, apparently?

_Update: Successfully signed a certificate with this change._

**Comment by sindarina** [comment]
It basically looks like the URL for the first signing step needs to be the same as for the second step?

**Comment by ghost** [comment]
I can confirm that, fixed by pull-request #33.

**Comment by diafygi** [comment]
Should be fixed now, thanks!
